### PR TITLE
 Tidy analyseNode

### LINKED ISSUE #1293: Tidy code in Analyser::AnalyserImpl::analyseNode
From #1291 the code could do with a little bit of tidying.
The tidying re-orders the analyser variables so a lot of the test results change. The tests results don't change semantically just the order in which variables are written out to file in.

### PR DESCRIPTION
**Pull request by hsorby**
Fixes #1293.

### CODE REVIEW

**Review by agarny — COMMENTED** [review]
I honestly can't see how this is tidying up the code. There are now more `if` statements and we even call `swapLeftAndRightChildren()` (7.23k times in our tests, i.e. ~94% of the time!). IOW, the code is now less efficient and, for me, more difficult to understand.

Otherwise, if you are to "tidy" that bit of code then you might also want to "tidy" the case of a piecewise statement (search for `childCount >= 2`).

Bottom line, if you guys are happy with this PR then feel free to force merge it.

**Review by nickerso — APPROVED** [review]
